torch/csrc/jit/tensorexpr/types.cpp

```diff
@@ -38,6 +42,13 @@ bool is_floating_point(const ScalarType& type) {
   return false;
 }
 
+bool is_signed(const ScalarType& type) {
```

Comment:
The generality of `is_signed` confused me a bit - given that it's only used on paths handling integrals anyway, I don't know if it's warranted. If we made it handle integrals only, it'd side-step the `is_c10_type` trick as well.